contracts/src/v0.8/vrf/dev/VRFCoordinatorV2_5.sol

```diff
@@ -367,8 +356,8 @@ contract VRFCoordinatorV2_5 is VRF, SubscriptionAPI, IVRFCoordinatorV2Plus {
   }
 
   function _getRandomnessFromProof(
-    Proof memory proof,
-    VRFTypes.RequestCommitmentV2Plus memory rc
+    Proof calldata proof,
```

Comment:
@jinhoonbang Any thoughts on why these parameters were having the memory modifier instead of calldata, if calldata is more efficient for read only purposes?

Reply:
In theory, `calldata` is always cheaper. But when I changed to `calldata` previously, it resulted in higher gas usage. I suspect it should be due to the way our tests are set up and not really because of the `calldata`. I didn't spend too much looking into this then as the contracts were more or less finalized and we were rushing to deploy.

Reply:
Yeah, I would think calldata is cheaper in terms of gas, in general.

Reply:
As we discussed on our call, I reverted all parameters to `memory` so we don't break the interface for the BatchCoordinator. The good news is that the contract size didn't break, but it is still very close to its limit.

With `calldata`:
| Contract                             | Size (kB) | Margin (kB) |
| VRFCoordinatorV2_5                   | 22.226    | 2.35        |
| VRFCoordinatorV2_5_Arbitrum          | 22.612    | 1.964       |
| VRFCoordinatorV2_5_Optimism          | 24.227    | 0.349       |

With `memory`:
| Contract                             | Size (kB) | Margin (kB) |
| VRFCoordinatorV2_5                   | 22.329    | 2.247       |
| VRFCoordinatorV2_5_Arbitrum          | 22.728    | 1.848       |
| VRFCoordinatorV2_5_Optimism          | 24.343    | 0.233       |